okhttp/src/main/java/okhttp3/internal/connection/RouteException.kt

```diff
@@ -13,36 +13,26 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package okhttp3.internal.connection;
+package okhttp3.internal.connection
 
-import java.io.IOException;
-
-import static okhttp3.internal.Util.addSuppressedIfPossible;
+import okhttp3.internal.Util.addSuppressedIfPossible
+import java.io.IOException
 
 /**
  * An exception thrown to indicate a problem connecting via a single Route. Multiple attempts may
  * have been made with alternative protocols, none of which were successful.
  */
-public final class RouteException extends RuntimeException {
-  private IOException firstException;
-  private IOException lastException;
-
-  RouteException(IOException cause) {
-    super(cause);
-    firstException = cause;
-    lastException = cause;
-  }
-
-  public IOException getFirstConnectException() {
-    return firstException;
-  }
+class RouteException internal constructor(val firstConnectException: IOException) :
+    RuntimeException(firstConnectException) {
+  var lastConnectException: IOException? = null
```

Comment:
This doesn't seem to be actually nullable. You can change the initializer to be `firstConnectException` so it always has a non-null value and remove the `init` block.